### ISSUE TITLE
Provide actionable error when Packer fails to connect to inactive HCP Packer registry. 

### ISSUE THREAD

**Issue by nywilken** [hashicorp/packer#11400]
When testing HCP Packer using the guide provided [here](https://learn.hashicorp.com/tutorials/packer/golden-image-with-hcp-packer?in=packer/cloud-production) I skipped over the enable registry step and encountered the following error. 

<img width="746" alt="Screen Shot 2021-10-15 at 11 13 58 AM" src="https://user-images.githubusercontent.com/1749304/142020277-06731c05-bf14-4505-9fea-94e94d73877a.png">

Only after reviewing the guide did I realize that I had to click the "Create Registry" button in order to start working with HCP Packer. In an effort to help users publishing to HCP Packer, or reading from HCP Packer via the [HCP datasources](https://www.packer.io/docs/datasources/hcp), this error information should be updated to provide an actionable error. So that users clearly understand what failed and how they can resolve it. 

This same error information should be made available to the Terraform HCP Packer provider - https://github.com/hashicorp/terraform-provider-hcp

**Comment by azr** [comment]
Okay, this was solved API side, and it feels like enough:


> **packerdev build ../../../datasource/hcp-packer-iteration/test-fixtures/template.pkr.hcl**
> Error: Datasource.Execute failed: error retrieving iteration from HCP Packer registry: [GET /packer/2021-04-30/organizations/{location.organization_id}/projects/{location.project_id}/images/{bucket_slug}/channels/{slug}][404] PackerService_GetChannel default  &{Code:5 Details:[] Error:No HPC Packer registry was found for this organization and project. To create a registry, please visit the HCP Packer tab in https://portal.cloud.hashicorp.com/. Message:No HPC Packer registry was found for this organization and project. To create a registry, please visit the HCP Packer tab in https://portal.cloud.hashicorp.com/.}


Closing for now. Tell me @nywilken if you feel differently about this one, we can reopen :D

**Comment by nywilken** [comment]
> Closing for now. Tell me @nywilken if you feel differently about this one, we can reopen :D

The error message from the API is enough for sure. But I was thinking that we could display it to the user a little better. More specifically, I was thinking that instead of just displaying the raw error message we could just display the message to make it easier for the user to digest. Something like 

```
packer build source.pkr.hcl
Error: HCP Packer Registry initialization failure

No HPC Packer registry was found for this organization and project. To create a registry, please visit the HCP 
Packer tab in https://portal.cloud.hashicorp.com/. Message:No HPC Packer registry was found for 
this organization and project. To create a registry, please visit the HCP Packer tab in 
https://portal.cloud.hashicorp.com/.
```

I created this [branch](https://github.com/hashicorp/packer/commit/be2c1b52c50ecffb4b1f074996b237973b8c598e#diff-8ba2dd8a6f6a51e7c5c68ccb5da25325102eec599e1ef4b9161169275ede6087R44) before the acceptance test refactor to get a sense of what I was thinking. Let me know if this is something we can achieve without too much lift.

**Comment by nywilken** [comment]
@azr I know that I mentioned cleaning up the API information from the error but seeing as there will be a larger piece of work around error handling for HCP Packer  I think we can call this one done as you mentioned above. Does that sound good to you? 

Or should we use it to track the large error handling work as well?

**Comment by azr** [comment]
I agree with you, we can call this one done, and follow the other large issue up.

Closing :D
